lib/webpacker/manifest.rb

```diff
@@ -13,32 +13,94 @@ def file_path
       Webpacker::Configuration.manifest_path
     end
 
-    def lookup(name)
+    # Converts the "name" (aka the pack or bundle name) to to the full path for use in a browser.
+    def asset_source(name)
+      output_path_or_url = Webpacker::Configuration.output_path_or_url
+      mapped_name = lookup(name)
+      "#{output_path_or_url}/#{mapped_name}"
+    end
+
+    # Converts the "name" (aka the pack or bundle name) to the possibly hashed name per a manifest.
+    #
+    # If Configuration.compile? then compilation is invoked the file is missing.
+    #
+    # Options
+    # throw_if_missing: default is true. If false, then nill is returned if the file is missing.
+    def lookup(name, throw_if_missing: true)
+      instance.confirm_manifest_exists
+
       if Webpacker::Configuration.compile?
-        compile_and_find!(name)
+        compile_and_find(name, throw_if_missing: throw_if_missing)
       else
-        find!(name)
+        # Since load_instance checks a `mtime` on the manifest for a non-production env before loading,
+        # we should always call this before a call to `find!` since one may be using
+        # `webpack -w` and a file may have been added to the manifest since Rails first started.
+        load_instance
+        raise Webpacker::FileLoader::FileLoaderError.new("Webpacker::Manifest.load must be called first") unless instance
+
+        return find(name, throw_if_missing: throw_if_missing)
       end
     end
 
+    # Why does this method exist? Testing? It's not in the README
     def lookup_path(name)
-      Rails.root.join(File.join(Webpacker::Configuration.public_path, lookup(name)))
+      Rails.root.join(File.join(Webpacker::Configuration.output_path, lookup(name)))
+    end
+
+    # Helper method to determine if the manifest file exists. Maybe Webpack needs to run?
+    # **Used by React on Rails.**
+    def exist?
```

Comment:
@gauravtiwari How about if I just remove the comment? There's 2 situations: 

1. No manifest.
2. Manifest is missing the files.